pkg/controllers/staticroute/staticroute_controller.go

```diff
@@ -47,41 +47,19 @@ type StaticRouteReconciler struct {
 	StatusUpdater common.StatusUpdater
 }
 
-func (r *StaticRouteReconciler) listStaticRouteCRIDs() (sets.Set[string], error) {
-	staticRouteList := &v1alpha1.StaticRouteList{}
-	err := r.Client.List(context.Background(), staticRouteList)
-	if err != nil {
-		log.Error(err, "failed to list StaticRoute CRs")
-		return nil, err
-	}
-
-	CRStaticRouteSet := sets.New[string]()
-	for _, staticroute := range staticRouteList.Items {
-		CRStaticRouteSet.Insert(string(staticroute.UID))
-	}
-	return CRStaticRouteSet, nil
-}
-
 func (r *StaticRouteReconciler) deleteStaticRouteByName(ns, name string) error {
-	CRPolicySet, err := r.listStaticRouteCRIDs()
-	if err != nil {
-		return err
-	}
 	nsxStaticRoutes := r.Service.ListStaticRouteByName(ns, name)
 	for _, item := range nsxStaticRoutes {
-		uid := util.FindTag(item.Tags, commonservice.TagScopeStaticRouteCRUID)
-		if CRPolicySet.Has(uid) {
-			log.Info("skipping deletion, StaticRoute CR still exists in K8s", "staticrouteUID", uid, "nsxStatciRouteId", *item.Id)
-			continue
+		log.Info("Deleting StaticRoute", "Namespace", ns, "Name", name, "nsxStaticRouteId", *item.Id)
+		vpcInfo, err := commonservice.ParseVPCResourcePath(*item.Path)
+		if err != nil {
+			return err
```

Comment:
Better to print the error log if a failure happens.

Reply:
Updated, thanks!